cmd/influx2cortex/main.go

```diff
@@ -5,51 +5,76 @@ import (
 	"errors"
 	"flag"
 	"fmt"
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
 	"os"
 
 	"github.com/grafana/dskit/flagext"
 	"github.com/grafana/dskit/services"
 	"github.com/grafana/influx2cortex/pkg/influx"
+	"github.com/grafana/influx2cortex/pkg/internalserver"
 	"github.com/weaveworks/common/logging"
 	"github.com/weaveworks/common/signals"
 )
 
 func main() {
-	conf := influx.ProxyConfig{}
+	logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stdout))
+	promRegisterer := prometheus.DefaultRegisterer
+
+	proxyConfig := influx.ProxyConfig{
+		Logger:     logger,
+		Registerer: promRegisterer,
+	}
+	internalServerConfig := internalserver.ServiceConfig{}
 
-	flag.BoolVar(&conf.EnableAuth, "auth.enable", true, "require X-Scope-OrgId header")
 	flagext.RegisterFlags(
-		&conf.HTTPConfig,
-		&conf.RemoteWriteConfig,
+		&proxyConfig,
+		&internalServerConfig,
 	)
 	flag.Parse()
 
-	service, err := influx.NewProxy(conf)
+	appServices := make([]services.Service, 0)
+
+	proxyService, err := influx.NewProxy(proxyConfig)
 	if err != nil {
 		_, _ = fmt.Fprintf(os.Stderr, "error instantiating influx2cortex proxy: %s\n", err)
 		os.Exit(1)
 	}
+	appServices = append(appServices, proxyService)
 
-	servCtx, cancelFn := context.WithCancel(context.Background())
-	defer cancelFn()
-
-	err = services.StartAndAwaitRunning(servCtx, service)
+	internalService, err := internalserver.NewService(internalServerConfig, logger)
 	if err != nil {
-		_, _ = fmt.Fprintf(os.Stderr, "error running influx2cortex: %s\n", err)
+		_, _ = fmt.Fprintf(os.Stderr, "error instantiating internal server: %s\n", err)
 		os.Exit(1)
 	}
+	appServices = append(appServices, internalService)
+
+	ctx, cancelFn := context.WithCancel(context.Background())
+	defer cancelFn()
+
+	for _, service := range appServices {
+		err = services.StartAndAwaitRunning(ctx, service)
+		if err != nil {
+			_, _ = fmt.Fprintf(os.Stderr, "error starting service: %s\n", err)
+			os.Exit(1)
+		}
+	}
 
 	// Look for SIGTERM and stop the server if we get it
-	handler := signals.NewHandler(logging.GoKit(service.Logger))
+	handler := signals.NewHandler(logging.GoKit(logger))
 	go func() {
 		handler.Loop()
-		service.StopAsync()
+		for _, service := range appServices {
+			service.StopAsync()
+		}
 	}()
 
-	err = service.AwaitTerminated(context.Background())
-	if err != nil && !errors.Is(err, context.Canceled) {
-		_, _ = fmt.Fprintf(os.Stderr, "error running influx2cortex: %s\n", err)
-		os.Exit(1)
+	for _, service := range appServices {
+		err = service.AwaitTerminated(context.Background())
+		if err != nil && !errors.Is(err, context.Canceled) {
+			_, _ = fmt.Fprintf(os.Stderr, "error in service: %s\n", err)
```

Comment:
instead of fmt.fprintf, use the logger

Reply:
Hah, I was wondering about that yesterday but decided to just stick w/ what was already there; but yea, will do!